### ISSUE TITLE
Closes after doing the Captcha

### ISSUE THREAD

**Issue by lordCuppa** [wolfswolke/aniworld_scraper#3]
**Describe the bug:**
I just found your aniworld scrapper and wanted to test it out, so I downloaded your code. My Problem is, that after solving the captcha nothing happens.

**Steps to Reproduce the bug:**

1. Downloaded the code
2. cmd.exe -> pip isntall -r requirements.txt
3. open start_app.py -> change anime_name = "fairy-tail" -> save
4. execute start_app.py

**Screenshots**
CMD WINDOW:
2022-12-08 15:47:44,101: INFO    ------------- AniWorldScraper v01-00-05 started ------------
2022-12-08 15:47:44,599: INFO    start_app -> We have this many seasons: 8
2022-12-08 15:47:45,029: INFO    start_app -> Season 1 has 48 Episodes.

DevTools listening on ws://127.0.0.1:53657/devtools/browser/f17d23db-eb8a-476a-b795-fc977ad2f529
[27424:18688:1208/154749.143:ERROR:device_event_log_impl.cc(215)] [15:47:49.143] USB: usb_device_handle_win.cc:1045 Failed to read descriptor from node connection: Ein an das System angeschlossenes Gerõt funktioniert nicht. (0x1F)
[27424:18688:1208/154749.145:ERROR:device_event_log_impl.cc(215)] [15:47:49.145] USB: usb_device_handle_win.cc:1045 Failed to read descriptor from node connection: Ein an das System angeschlossenes Gerõt funktioniert nicht. (0x1F)
2022-12-08 15:48:01,919: ERROR   start_app -> ----------
2022-12-08 15:48:01,919: ERROR   start_app -> Exception: can only concatenate str (not "NoneType") to str
2022-12-08 15:48:01,919: ERROR   start_app -> ----------

![grafik](https://user-images.githubusercontent.com/98316739/206477674-5d7483c4-04e3-4491-bb8b-2cf0da50b4bc.png)

NOTEPAD ++:
anime_name = "fairy-tail"
anime_url = "https://aniworld.to/anime/stream/{}/".format(anime_name)
![grafik](https://user-images.githubusercontent.com/98316739/206478942-be6051ee-4626-4329-8602-0380212cdc1f.png)

**Additional Info**
Hopefully it is a fixable problem, because that would be some really nice program. Also, would it be interesting if I cloud select a specific season to download.

**Comment by wolfswolke** [comment]
@lordCuppa Problem found. One of aniworlds hosts got taken down. I will add a logic to the code that searches for Vidoza and doesnt just count up. In the mean time do this: go to search_for_links.py -> on line 33 change 4 to 3.
I will fix the Problem now but this should allow you to download again.  (The 3 just means the 4th button under the video.)
![image](https://user-images.githubusercontent.com/34097612/206484160-1c1fd699-76da-4e65-b25b-63859f9fe40a.png)
Please test it again and tell me if that fixes the Problem for you ^^

**Comment by lordCuppa** [comment]
@wolfswolke Thx. I will try it out. Are also planing in adding VOE (because I download 98% over VOE).

**Comment by lordCuppa** [comment]
![grafik](https://user-images.githubusercontent.com/98316739/206485946-c2da8471-3244-4943-a83a-c4b7fd47e054.png)
![grafik](https://user-images.githubusercontent.com/98316739/206486060-7d230961-cb15-4f2c-9d23-0a3d804e2bfc.png)

**Comment by wolfswolke** [comment]
@lordCuppa 
I never looked at VOE but i can add a streaming service switch to the TODO list.
And for the Problem you have now... looks like that python is not allowed to write to that location.
Can you try to move the howl folder to somewhere where you know you have R&W permission?

**Comment by lordCuppa** [comment]
@wolfswolke 
At first, I had it in the Windows Video folder, and now I tried it on the Desktop. But same issue.
Could there be some problem with the python installation? (I'm starting to learn python in my freetime)

**Comment by wolfswolke** [comment]
@lordCuppa 
I added a Check for Read and write perms. Please reclone the Repo and try again.
You need to set logger to debug. (start_app.py -> Line 50 -> replace info with debug)
I am also setting up a VM with a fresh install of python and ill test there if i can reproduce your issue.
And aslong as you didnt change alot with your py install it should work. I had problems before with the MS-Store version but that may be a me problem ^^"

**Comment by lordCuppa** [comment]
@wolfswolke I will try it later. Yeah I also have the MS Store version of python. There I for example i had the problem in geany that I needed to change the default debug command from py to python (or the other way around).

**Comment by lordCuppa** [comment]
@wolfswolke unfortunately it is still not working, but on the other hand it showed something different in the cmd window.
![grafik](https://user-images.githubusercontent.com/98316739/206876661-0301395a-f65d-494d-b5fa-18e4eab7960a.png)

**Comment by wolfswolke** [comment]
@lordCuppa I tried getting your error aswell but i cant get it to trigger.
Is it possible you had no Network? The error just says it could not connect to the Server and i tried it with Multiple Devices no issue. Maby try rebooting your Machine? Im not sure how you got that xD

**Comment by wolfswolke** [comment]
Closed no response.

**Comment by lordCuppa** [comment]
I'm really sorry for not responding. But even after rebooting I still get the same error.
![Screenshot 2023-01-03 115940](https://user-images.githubusercontent.com/98316739/210354407-240af4ab-d5f2-48da-8634-c2ebe8e88e5b.png)

**Comment by wolfswolke** [comment]
You are using a outdated version. Please try again with the latest release.

**Comment by lordCuppa** [comment]
I downloaded the latest version and now it doesn't start at all.
What I did: After downloading extract the zip -> open cmd: pip install -r requierements.txt -> open start_app.py and changed the anime_name variable to "vampire-princess-miyu".
After that I double click the start_app.py script but nothing happens.

**Comment by wolfswolke** [comment]
you dont double click a python script. you allways go to cmd and run it from there because if it fails you dont get a log any other way.

**Comment by lordCuppa** [comment]
So now I started it over cmd with: python start_app.py and now I get a response.

![grafik](https://user-images.githubusercontent.com/98316739/210428849-c44de327-b61f-40fe-84d5-e23bd4dc51be.png)
![grafik](https://user-images.githubusercontent.com/98316739/210429754-7786362c-d666-4959-9387-813583a788c0.png)

I will let it run and inform you about the results after it finished.

**Comment by lordCuppa** [comment]
Result: It downloaded without any problems, except that it only downloaded the first season.
Therefore, I looked up your README file and checked season override, but that had the right value: <0>
![grafik](https://user-images.githubusercontent.com/98316739/210434944-d39aed89-fab8-434a-b08b-535b82891892.png)

![grafik](https://user-images.githubusercontent.com/98316739/210434534-38e2fa9f-1966-4e3f-b968-bd5fda62f927.png)

But I am really happy that it is working at all.

**Comment by wolfswolke** [comment]
@lordCuppa Found the problem. The Episode you tried to download didn't have some streaming host online at that time.
That meant that Vidoza was not the third button but the second one.
I added some Logic so it can switch buttons.
Please try again if you can get it to download everything. You can also test the Ublock Origin extension with it if you want.
It makes it more stable and faster because it now doesn't load all the Ads anymore.

**Comment by lordCuppa** [comment]
@wolfswolke I just tested the new version, and it works without problems. Even the big error messages I got before (with: USB: usb_device_handle…) are gone. The only issue that appears is when the vidoza stream doesn't load correctly.

Furthermore, have I tried to use the new uBlock Origin feature, but that's not working.
![grafik](https://user-images.githubusercontent.com/98316739/211350442-18639e9e-9933-4639-8e26-381be09ab657.png)
I downloaded the full uBlock extension with your provided link from the read me file and put it in the folder. Have I done something incorrect?
![grafik](https://user-images.githubusercontent.com/98316739/211350814-96928805-b7b2-421a-9ed6-3829fd6e411b.png)
Really appreceate your work. THX
